### DISCUSSION TITLE
keep the non breaking spaces on the HTML output

### DISCUSSION THREAD

**Discussion by Beurt** [marp-team/marp#475] [Q&A] [answered]
Hi,

I use French. In French, we use **non-breaking spaces** (`&nbsp;` in HTML) very often (e.g. before every double punctuation, such as semicolons).

There are Unicode characters for non-breaking spaces ([yes there are multiple sorts :-)](https://en.wikipedia.org/wiki/Non-breaking_space#Width_variation)). Unfortunately, Marpit removes them from the HTML output.

input:

```md
There is a non-breaking space before the exclamation mark !
```

HTML output (it is gone, nor replaced with `&nbsp;` which would have been good too):

```html
<p>There is a non-breaking space before the exclamation mark !</p>
```

Does anyone know how to keep the non-breaking space with Marpit (usually, I use marp-cli)?

**Comment by yhatt** [comment, marked as answer]
It was not reproduced. At least to me, it looks like non-breaking spaces in Markdown are preserved also in rendered HTML.

<p><img width="769" alt="nbsp" src="https://github.com/marp-team/marp/assets/3993388/8217bf46-5bbb-4097-bd0b-44a3b49dd82e" /></p>

Unicode `0xa0` (=no-break space) is visualized by VS Code. You can also inspect `&nbsp;` in the exported HTML.

**Reply by Beurt** [reply]
I am very sorry, you are true, it appears that it is my editor that is removing the non breaking spaces!

Thank you for your quick answer.
